### PR TITLE
Update imAuthSigninsMultipleCountries.yaml

### PR DESCRIPTION
**Pull request by azurekid**
Required items, please complete
   
   Change(s):
   - Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml
   
   Reason for Change(s):
   - #8329

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml [hunk 1]
```diff
@@ -23,20 +23,24 @@ query: |
...
-  | where EventType=='Logon' and EventResult=='Success'
+  | where EventType == 'Logon' and EventResult == 'Success'
   | where isnotempty(SrcGeoCountry)
-  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), Vendors=make_set(EventVendor), Products=make_set(EventProduct)
-    , NumOfCountries = dcount(SrcGeoCountry)
-    by TargetUserId, TargetUsername, TargetUserType
+  | summarize
+      StartTime        = min(TimeGenerated)
+      , EndTime        = max(TimeGenerated)
+      , Vendors        = make_set(EventVendor)
```
Comment (devikamehra) [inline Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml:31]: All make_set should have maximum allowed value.

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml [hunk 2]
```diff
@@ -23,20 +23,24 @@ query: |
...
+      , NumOfCountries = dcount(SrcGeoCountry)
+      , Countries      = make_set(SrcGeoCountry)
+      by TargetUserId, TargetUsername, TargetUserType
   | where NumOfCountries >= threshold
   | extend timestamp = StartTime, AccountCustomEntity = TargetUsername
-
 entityMappings:
   - entityType: Account
     fieldMappings:
       - identifier: FullName
```
Comment (devikamehra) [inline Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml:41]: No longer supports the FullName. Please replace with Name and UPNSuffix

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml (hunk 1 above)
Comment (azurekid) [reply Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml]: Added a max value.

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml (hunk 2 above)
Comment (azurekid) [reply Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml]: Added both Name and UPNSuffix as Strongest identifiers according to the documentation.
https://learn.microsoft.com/en-us/azure/sentinel/entities-reference

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml [hunk 3]
```diff
@@ -23,20 +23,28 @@ query: |
...
   - entityType: Account
     fieldMappings:
-      - identifier: FullName
-        columnName: AccountCustomEntity
-version: 1.2.2
+      - identifier: Name
+        columnName: Account
+  - entityType: Account
+    fieldMappings:
+      - identifier: UPNSuffix
```
Comment (devikamehra) [inline Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml:45]: Entity is not required to be repeated. Please include the UPN identifier in the same Entity as the Name.
Also, you would have to split the value of TargetUsername on "@" to get the Name and UPNSuffix.

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml [hunk 4]
```diff
@@ -23,20 +23,28 @@ query: |
...
+  | summarize
+      StartTime        = min(TimeGenerated)
+      , EndTime        = max(TimeGenerated)
+      , Vendors        = make_set(EventVendor, 128)
+      , Products       = make_set(EventProduct, 128)
+      , NumOfCountries = dcount(SrcGeoCountry)
+      , Countries      = make_set(SrcGeoCountry, 128)
+      by TargetUserId, TargetUsername, TargetUserType
   | where NumOfCountries >= threshold
   | extend timestamp = StartTime, AccountCustomEntity = TargetUsername
```
Comment (devikamehra) [inline Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml:37]: This line is no longer required. Rather split the TargetUsername here.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml (hunk 3 above)
Comment (azurekid) [reply Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml:45]: I just thought to simply pick up an issue of an existing rule and add the requested field. Didn't expect so many other changes o  a rule that is already in production.

Will add the requested changes later tonight

**Comment by v-atulyadav** [comment]
Hi @azurekid, please act on @devikamehra's comments. Thanks

**Review by azurekid — COMMENTED** [review]
Updated split of TargetUsername

**Review by devikamehra — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**